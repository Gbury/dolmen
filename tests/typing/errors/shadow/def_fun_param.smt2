(set-logic ALL)
(define-fun f ((x Int) (x Int)) Bool (= x x))
(check-sat)
(exit)
