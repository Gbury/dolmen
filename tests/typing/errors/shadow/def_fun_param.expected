File "tests/typing/errors/shadow/def_fun_param.smt2", line 2, character 23-30:
2 | (define-fun f ((x Int) (x Int)) Bool (= x x))
                           ^^^^^^^
Error Parameter `x` was already bound at line 2, character 15-22
      as another parameter of the same function definition
Hint: Shadowing between parameters in the same function definition in
  SMT-LIB2 has extremely un-inuitive semantics, and lead to partially
  specified functions. It definitely doesn't do what you expect and such
  definitions are therefore strongly discouraged. For more information, read
  https://github.com/SMT-LIB/SMT-LIB-2/issues/36
